src/SignalR/docs/specs/TransportProtocols.md

```diff
@@ -12,19 +12,65 @@ A transport is required to have the following attributes:
+* If the servers minimum supported protocol version is greater than the version requested by the client it will send an error response and close the connection
+* If the server supports the request version it will respond with the requested version
+* If the requested version is greater than the servers largest supported version the server will respond with its largest supported version
+The client may close the connection if the "negotiateVersion" in the response is not acceptable.
 
+The content type of the response is `application/json` and is a json payload containing properties to assist the client in establishing a persistent connection.
+
+### Version 1
+
+When the server and client agree on version 1 the server response will include a "connectionToken" property in addition to the "connectionId" property. The value of the "connectionToken" property will be used in the "id" query string for the HTTP requests described below, this value should be kept secret.
+
+A successful negotiate response will look similar to the following payload:
+  ```json
+  {
+    "connectionToken":"05265228-1e2c-46c5-82a1-6a5bcc3f0143",
+    "connectionId":"807809a5-31bf-470d-9e23-afaee35d8a0d",
+    "negotiateVersion":1,
+    "availableTransports":[
+      {
+        "transport": "WebSockets",
+        "transferFormats": [ "Text", "Binary" ]
+      },
+      {
+        "transport": "ServerSentEvents",
+        "transferFormats": [ "Text" ]
+      },
+      {
+        "transport": "LongPolling",
+        "transferFormats": [ "Text", "Binary" ]
+      }
+    ]
+  }
+  ```
+
+  The payload returned from this endpoint provides the following data:
+
+  * The `connectionToken` which is **required** by the Long Polling and Server-Sent Events transports (in order to correlate sends and receives).
+  * The `connectionId` which is the id by which other clients can refer to it.
+  * The `negotiateVersion` which is the negotiation protocol version being used between the server and client.
+  * The `availableTransports` list which describes the transports the server supports. For each transport, the name of the transport (`transport`) is listed, as is a list of "transfer formats" supported by the transport (`transferFormats`)
+
+### Version 0
+
+When the server and client agree on version 0 the server response will include a "connectionId" property that is used in the "id" query string for the HTTP requests described below.
+
+A successful negotiate response will look similar to the following payload:
   ```json
   {
     "connectionId":"807809a5-31bf-470d-9e23-afaee35d8a0d",
+    "negotiateVersion":0,
```

Comment:
Version `0` won't have a version in the response at all, right?

Reply:
Client could receive this if talking to a new server

Reply:
It feels like the `negotiateVersion` (or whatever it's called) should just not be present in that case. _Our_ clients don't bork if extra properties are present but others might. We can't guarantee that other clients will be compatible, but we can work to avoid breaking them when we can.

Would also be good to put an explicit thing in about clients being required to accept extra JSON properties.

Reply:
I think as a general rule, we should require client to ignore unknown properties in payloads and what order the properties arrive in.

While it's possible for the server to use the requested and/or selected `negotiateVersion` to generate exactly the payload the client expects, I'd rather not go through that extra effort. There also may come a time when we want to add a property to a payload without incrementing and special-casing yet another the protocol version.

Reply:
Lets decide on this quickly, now that the PR is out for 3.1

Reply:
You know my position, but I certainly won't push back hard if people feel we should omit the `negotiateVersion` in this case.

Reply:
> I think as a general rule, we should require client to ignore unknown properties in payloads and what order the properties arrive in.

I do entirely agree with this. It's just a new requirement we're putting on external clients that they may not have been prepared for. Leaving the "version 0" responses untouched protects us from that problem while allowing "version 1 and up" to add the *requirement* that clients ignore extra fields.

Buuuutt.... I don't care a lot. It should be safe to leave this and patch it out again if a client does come along that requires us to strip this property.

Leave it in. Let's land this ✈.